src/main/java/com/simibubi/create/content/kinetics/chainConveyor/ChainConveyorRenderer.java

```diff
@@ -100,10 +111,22 @@ private void renderBox(ChainConveyorBlockEntity be, PoseStack ms, MultiBufferSou
 			physicsData.modelKey = key;
 		}
 
+		//More Safety Checks
+		ResourceLocation modelKey = physicsData.modelKey;
+		if (modelKey == null)
```

Comment:
As Tropheus said this is the only check that should be necessary and the others should be removed. Do we need a similar check in the ChainConveyorVisual?

Also can we move it up in the method to skip the expensive light lookups if we're going to early out anyway?